cluster-autoscaler/core/scaledown/planner/planner.go

```diff
@@ -96,19 +94,13 @@ func New(context *context.AutoscalingContext, processors *processors.Autoscaling
 // UpdateClusterState needs to be periodically invoked to provide Planner with
 // up-to-date information about the cluster.
 // Planner will evaluate scaleDownCandidates in the order provided here.
-func (p *Planner) UpdateClusterState(podDestinations, scaleDownCandidates []*apiv1.Node, as scaledown.ActuationStatus, pdbs []*policyv1.PodDisruptionBudget, currentTime time.Time) errors.AutoscalerError {
+func (p *Planner) UpdateClusterState(podDestinations, scaleDownCandidates []*apiv1.Node, as scaledown.ActuationStatus, currentTime time.Time) errors.AutoscalerError {
 	p.latestUpdate = currentTime
 	p.actuationStatus = as
-	p.remainingPdbTracker = pdb.NewBasicRemainingPdbTracker()
-	err := p.remainingPdbTracker.AddPdbs(pdbs)
-	if err != nil {
-		p.CleanUpUnneededNodes()
```

Comment:
Shouldn't we now call this in `static_autoscaler.go` when the same call fails? Or does that already happen because we're failing the whole loop?

Reply:
I think this code was actually dead. The only way for this check to trigger would be for the PDB Lister to return PDBs with ill configured selectors, which should not be possible.